behave/__main__.py

```diff
@@ -180,7 +181,33 @@ def main(args=None):
     :return: 0, if successful. Non-zero, in case of errors/failures.
     """
     config = Configuration(args)
-    return run_behave(config)
+
+    rclass = Runner
+    if getattr(config, 'proc_count'):
+        try:
+            from behave.runner_mp import MultiProcRunner_Feature, MultiProcRunner_Scenario
+            pelem = getattr(config, 'parallel_element', False)
+            if not pelem:
+                print ("INFO: Without giving --parallel-element, defaulting to 'scenario'...")
+                pelem = 'scenario'
+
+            if pelem == 'scenario':
+                rclass = MultiProcRunner_Scenario
+            elif pelem == 'feature':
+                rclass = MultiProcRunner_Feature
+            else:
+                print ("ERROR: When using --processes, --parallel-element"
+                    " option must be set to 'feature' or 'scenario'. You gave "
+                    "'%s', which isn't valid." % pelem)
+                return 1
+
+        except ImportError as e:
+            print ("DEBUG: import error: %s" % e)
+            print ("ERROR: Cannot import multiprocessing module."
+            " If you're on python2.5, go get the backport")
+            return 1
```

Comment:
OMG, how old is that code!?! :astonished: Do we really need to support Python 2.5?

Reply:
@bittner b500c39535b4c17605ec , from 2013!
Still, it would make sense if run on some py interpreter that doesn't support multiprocessing today.

Reply:
@xrg out of curiosity, what interpreter would that be? And if so, couldn't they just install https://pypi.org/project/multiprocessing/ (very outdated) instead of having a fork in the code to address a very unlikely scenario, plus extra nesting?

Reply:
TBH, i don't really know/remember why that 'if' would be needed for multiprocessing. I had just refactored that old code years ago. And /thought/ that there were still some mis-behaving interpreters around. Removing the conditional could classify as an improvement, agreed.

Reply:
Agreed then. Ps. can you merge upstream/master down and fix your conflicts?

Reply:
I

> OMG, how old is that code!?! 😲 Do we really need to support Python 2.5?

Python 2.5 support we shouldn't be worried now. Fix it by removing the if condition and at least get it running in 2.7 would make sense.